packages/@aws-cdk/aws-cloudfront/lib/key-group.ts

```diff
@@ -0,0 +1,73 @@
+import { Names, Resource, Token } from "@aws-cdk/core";
+import { Construct } from 'constructs';
+import { CfnKeyGroup } from './cloudfront.generated';
+
+/**
+ * Represents a Key Group
+ */
+export interface IKeyGroup {
+  /**
+   * The ID of the key group.
+   * @attribute
+   */
+  readonly keyGroupId: string;
+}
+
+/**
+ * Properties for creating a Key Group
+ */
+export interface KeyGroupProps {
+  /**
+   * A name to identify the key group.
+   * @default - generated from the `id`
+   */
+  readonly keyGroupName?: string;
+  
+  /**
+   * A comment to describe the key group.
+   * @default - no comment
+   */
+  readonly comment?: string;
+
+  /**
+   * A list of the identifiers of the public keys in the key group.
+   */
+  readonly items?: string[];
+}
+
+/**
+ * A Key Group configuration
+ * 
+ * @resource AWS::CloudFront::KeyGroup
+ */
+export class KeyGroup extends Resource implements IKeyGroup {
+
+  public static fromKeyGroupId(scope: Construct, id: string, keyGroupId: string): IKeyGroup {
+    return new class extends Resource implements IKeyGroup {
+      public readonly keyGroupId = keyGroupId;
+    }(scope, id);
+  }
+
+  public readonly keyGroupId: string;
+
+  constructor(scope: Construct, id: string, props: KeyGroupProps = {}) {
+    super(scope, id, {
+      physicalName: props.keyGroupName,
+    });
+
+    const keyGroupName = props.keyGroupName ?? Names.uniqueId(this);
+    if (!Token.isUnresolved(keyGroupName) && !keyGroupName.match(/^[\w-]+$/i)) {
```

Comment:
For both this and PublicKey, is there a source for the regex? I don't see anything in the docs explicitly.

Reply:
For KeyGroups I saw example on this page: https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-cloudfront-keygroup.html

> Ref
> When you pass the logical ID of this resource to the intrinsic Ref function, Ref returns the ID of the key group. For example: e9fcd3cf-f3f4-4b61-bd85-9ba9e091b309. 

However, for PublicKey ref is different... https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-cloudfront-publickey.html. Is there a helper method for these kinds of logical IDs?

> Ref
> When you pass the logical ID of this resource to the intrinsic Ref function, Ref returns the ID of the public key. For example: K36X4X2EO997HM.

So... yeah definitely needs a different format. Any suggestions on how to handle this?

Reply:
Both of those are the resulting ref/ID of the resource _after_ it's been created, not before. In lieu of any specific guidance/restrictions on the name, I would opt to just take what the user gives us wholesale. We can always add validation -- once known later.

Reply:
Sounds good. I'll remove this then (for now).